sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/regexpExpressions.scala

```diff
@@ -178,6 +180,90 @@ case class Like(left: Expression, right: Expression, escapeChar: Char)
   }
 }
 
+/**
+ * Optimized version of LIKE ALL, when all pattern values are literal.
+ */
+abstract class LikeAllBase extends UnaryExpression with ImplicitCastInputTypes with NullIntolerant {
+
+  protected def patterns: Seq[UTF8String]
+
+  protected def isNotDefined: Boolean
+
+  override def inputTypes: Seq[DataType] = StringType :: Nil
+
+  override def dataType: DataType = BooleanType
+
+  override def nullable: Boolean = true
+
+  private lazy val hasNull: Boolean = patterns.contains(null)
+
+  private lazy val cache = patterns.filterNot(_ == null)
+    .map(s => Pattern.compile(StringUtils.escapeLikeRegex(s.toString, '\\')))
+
+  override def eval(input: InternalRow): Any = {
+    val exprValue = child.eval(input)
+    if (exprValue == null) {
+      null
+    } else {
+      val allMatched = if (isNotDefined) {
+        !cache.exists(p => p.matcher(exprValue.toString).matches())
+      } else {
+        cache.forall(p => p.matcher(exprValue.toString).matches())
+      }
+      if (allMatched && hasNull) {
+        null
+      } else {
+        allMatched
+      }
+    }
+  }
+
+  override def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = {
+    val eval = child.genCode(ctx)
+    val patternClass = classOf[Pattern].getName
+    val javaDataType = CodeGenerator.javaType(child.dataType)
+    val pattern = ctx.freshName("pattern")
+    val allMatched = ctx.freshName("allMatched")
+    val valueIsNull = ctx.freshName("valueIsNull")
+    val valueArg = ctx.freshName("valueArg")
+    val patternHasNull = ctx.addReferenceObj("hasNull", hasNull)
```

Comment:
It's a boolean contant. We can change the generated code based on it.

Reply:
OK